apps/graphql-playground/src/components/GqlPlayground.tsx

```diff
@@ -87,10 +90,36 @@ function GqlPlayground(props: GqlPlaygroundProps) {
 
   const settings = { 'editor.theme': 'light' };
 
+  useEffect(() => {
+    const unsubscribe = store.subscribe(() => {
+      const query: string = getQuery(store.getState());
+      const result = query.toLowerCase().includes('collection');
+      if (hasCollection !== result) {
+        setHasCollection(result);
+      }
+    });
+    return unsubscribe;
+  }, [hasCollection]);
```

Comment:
`hasCollection` should be removed from the dependencies here. We currently unsubscribe and resubscribe whenever `hasCollection` changes. We actually only want to subscribe on mount and unsubscribe on unmount.